databuilder/databuilder/task/search/search_metadata_to_elasticsearch_task.py

```diff
@@ -0,0 +1,132 @@
+        # iterate through records
+        while record:
+            record = self.transformer.transform(record)
+            if not record:
+                # Move on if the transformer filtered the record out
+                record = self.extractor.extract()
+                continue
+            yield self.to_document(document_mapping=self.document_mapping,
+                                   metadata=record,
+                                   index=self.elasticsearch_new_index).to_dict(True)
+            record = self.extractor.extract()
+
+    def _delete_old_index(self, connection: Connections, document_index: Index) -> None:
+            alias_updates = [
+            {"add": {"index": document_index._name, "alias": self.elasticsearch_alias}}
+            ]
+            for index_name in connection.indices.get_alias():
+                if index_name.startswith(f"{self.elasticsearch_alias}_"):
+                    if index_name != document_index._name:
+                        LOGGER.info(f"Deleting index old {index_name}")
+                        alias_updates.append({"remove_index": {"index": index_name}})
+            connection.indices.update_aliases({"actions": alias_updates})
+
+    def run(self) -> None:
+        LOGGER.info('Running search metadata to Elasticsearch task')
+        try:
+            # create connection
+            connections.add_connection('default', self.elasticsearch_client)
+            connection = connections.get_connection()
+            
+            # health check ES
+            health = connection.cluster.health()
+            status = health["status"]
+            if status not in ("green", "yellow"):
+                msg = f"Elasticsearch healthcheck failed: {status}"
+                LOGGER.error(msg)
+                raise Exception(msg)
+
+            # extract records from metadata store
+            record = self.extractor.extract()
+            
+            # create index
+            index = self.index_class(name=self.elasticsearch_new_index)
+            
+            # publish search metadata to ES
+            cnt = 0
+            for success, info in parallel_bulk(connection,
+                                               self.generator(record=record),
+                                               raise_on_error=False):
+                if not success:
+                    LOGGER.warn(f"There was an error while indexing a document to ES: {info}")
+                else:
+                    cnt += 1
+                if cnt == self.elasticsearch_batch_size:
+                    LOGGER.info(f'Published {str(cnt)} records to ES')
+
+            # delete old index
+            self._delete_old_index(connection=connection,
+                                   document_index=index)
+
+            LOGGER.info("Elasticsearch Indexing completed")
+        finally:
+            self._closer.close()
+
+    def get_scope(self) -> str:
+        return 'task.search_metadata_to_elasticsearch'
```

Comment:
maybe extract this as a task_name variable for others to use. Also nit i think maybe something like "elasticsearch_dsl_publisher" may be a better name.

Reply:
does it matter if it's dls or not? :P + this would make it confusing as this is a task with publisher in name while we have concept of publishers in amundsen

Reply:
its not just a publisher, wouldn't want tog et it confused with actual publishers